File: php/src/Behavioral/Mediator/DvdUppercaseTitle.php
<?

#//DvdUpcaseTitle - Two of Two Concrete Colleagues or Mediatees

require_once 'DvdTitle.php';


class DvdUppercaseTitle extends DvdTitle
{

	public $uppercasetitle;
	public $mediator;

	function __construct($title,$mediator)
	{
		$this->setTitle( is_object($title) ? $title->getTitle() : $title);
		$this->mediator = $mediator;
		$this->resetTitle(NULL);
		$this->mediator->uppercasetitle  = $this;
	}


	function resetTitle($title)
	{
		if($title)
			$this->setTitle($title) ;
		$this->uppercasetitle = strtoupper( $this->getTitle());
	}

	function setSuperTitleUpcase()
	{
		$this->setTitle( $this->uppercasetitle );
		$this->dvdMediator->changeTitle($this);       
	}
}

File: ruby/src/Behavioral/Mediator/DvdUppercaseTitle.rb

#//DvdUpcaseTitle - Two of Two Concrete Colleagues or Mediatees

require "Behavioral/Mediator/DvdTitle"

class DvdUppercaseTitle < DvdTitle

	attr_accessor :uppercasetitle, :title

	def initialize(title, dvdMediator)
		@uppercasetitle = nil
		if title.is_a? DvdTitle
			setTitle( title.getTitle() )
		else 
			setTitle( title )
		end
		resetTitle()
		@dvdMediator = dvdMediator
		@dvdMediator.setUpcase(self)
	end

	def resetTitle(title = nil )
		if title
 			setTitle(title) 
		end
		setUpcaseTitle(getTitle().upcase())
	end

	def setSuperTitleUpcase()
 		setTitle(getUpcaseTitle())
		@dvdMediator.changeTitle(self)
	end

	def getUpcaseTitle()
 		return @uppercasetitle
	end

	def setUpcaseTitle (upcaseTitle)
 		@uppercasetitle = upcaseTitle
	end

end
